content/docs/reference/buildpack-api.md

```diff
@@ -213,20 +230,55 @@ version = "10.x"
 
 [requires.metadata]
 version-source = ".nvmrc"
+```
 
-[[or]]
+As always, the buildpack `provides` `node`. In this particular case, a version of `node` (`10.x`) is being `requested` in a configuration file (`.nvmrc`), and the buildpack chooses to add an additional piece of metadata (`version-source`), so that it can understand where that request came from.
 
-[[or.provides]]
+#### 3. Two Versions Requested
+The buildpack sees two configuration files (e.g. `.nvmrc` and `buildpack.yml`) that request two different versions of the `node` dependency. One way of dealing with that, would be writing both `requests` to the build plan, and reconciling which version(s) it will provide during the `build` phase. It can provide two different `requires`:
+
+```
+[[provides]]
 name = "node"
 
-[[or.requires]]
+[[requires]]
+name = "node"
+version = "10.x"
+
+[requires.metadata]
+version-source = ".nvmrc"
+
+[[requires]]
 name = "node"
 version = "12.x"
 
-[or.requires.metadata]
+[requires.metadata]
 version-source = "buildpack.yml"
 ```
 
+As always, the buildpack provides `node`. In this case, it requests two different things: `node v10.x` and `node v12.x`. During the `build` phase, it will analyze the two `requires`, and decide which (or both) to follow. Alternatively, it could have resolved the appropriate version in the `detect` phase, and only written the corresponding entry to the build plan.
```

Comment:
> decide which (or both) to follow

Can a node-engine buildpack provide two different versions of node? It seems so from what's presented here but maybe I am misunderstanding? This might be good to clarify.

> it could have resolved the appropriate version in the `detect` phase

I would love to know how this is done! This info might be helpful to complete the picture.